core/adapters/multiply_test.go

```diff
@@ -75,15 +75,15 @@ func TestMultiply_Unmarshal(t *testing.T) {
 func TestMultiply_Perform(t *testing.T) {
 	tests := []struct {
 		name  string
-		Times *big.Float
+		Times decimal.Decimal
 		json  string
 		want  string
 	}{
-		{"by 100", big.NewFloat(100), `{"result":"1.23"}`, "123"},
-		{"float", big.NewFloat(100), `{"result":1.23}`, "123"},
-		{"negative", big.NewFloat(-5), `{"result":"1.23"}`, "-6.15"},
-		{"no times parameter", nil, `{"result":"3.14"}`, "3.14"},
-		{"zero", big.NewFloat(0), `{"result":"1.23"}`, "0"},
+		{"by 100", decimal.NewFromInt(100), `{"result":"1.23"}`, "123"},
+		{"float", decimal.NewFromInt(100), `{"result":1.23}`, "123"},
+		{"negative", decimal.NewFromInt(-5), `{"result":"1.23"}`, "-6.15"},
+		{"no times parameter", decimal.NewFromInt(1), `{"result":"3.14"}`, "3.14"},
+		{"zero", decimal.NewFromInt(0), `{"result":"1.23"}`, "0"},
```

Comment:
Can we see a test case for the extremely large values that were being incorrectly calculated before?

I.e. A case that fails before this fix, and passes afterwards.

Reply:
Added